src/main/java/com/minecolonies/coremod/entity/ai/util/ChatSpamFilter.java

```diff
@@ -64,22 +65,44 @@ public ChatSpamFilter(final EntityCitizen worker)
      *
      * @param chat the Item Name
      */
-    public void requestWithoutSpam(@NotNull final String chat)
+    public void requestWithoutSpam(@NotNull final TextComponentBase chat)
     {
         talkWithoutSpam("entity.miner.messageNeedBlockAndItem", chat);
     }
 
     /**
+     * Get a describing string of an Object array.
+     * @param chat the object array.
+     * @return the describing string.
+     */
+    public String getStringOfChat(final Object[] chat)
+    {
+        StringBuilder tempString = new StringBuilder();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'tempString' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)